controllers/terminal/controllers/terminal_controller.go

```diff
@@ -48,14 +69,167 @@ type TerminalReconciler struct {
 // - https://pkg.go.dev/sigs.k8s.io/controller-runtime@v0.12.1/pkg/reconcile
 func (r *TerminalReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
 	_ = log.FromContext(ctx)
+	terminal := &terminalv1.Terminal{}
```

Comment:
add Finalizer for CRD ?  you can using controller sdk https://github.com/labring/endpoints-operator/blob/main/library/controller/controller.go . auto add Finalizer